### PR TITLE
Refactor/Refactor DB attributes

### LINKED ISSUE #114: Feature/Refactor attributes in database
### Attribute to expose
- [x] Add `hasVoted` to `userRoutesIndex`
- [x] Add `hasVoted` to `gymLocationIndex`
- [x] Refactor `getRoutesByGym` and `getRoutesByUser`
- [x] Update swagger

### Attribute to remove
- [x] Remove `hasGraded` from all
- [x] Remove `voteCount`
- [x] Update swagger

### PR DESCRIPTION
**Pull request by yarkhinephyo**
### Summary
Resolves #114 by changing adding `hasVoted` to responses of `GET /route/all` and `GET /route/user`. As those are public endpoints, if no authorization token is provided, `hasVoted` will return false by default.

### Notes
Updated google docs and swagger.
No changes in Postman.
Removed `hasGraded` from `POST /route/details`

### CODE REVIEW

File: lambda_backend/route_microservice/src/getRoutesByUser.ts [hunk 1]
```diff
@@ -58,18 +62,37 @@ const getRoutesByUser: Handler = async (event: GetRoutesByUserEvent) => {
...
+  let username = '';
+  if (Authorization) {
+    // Only utilizes token for username, no enforcing authorization as endpoint is GET & public
+    ({ username } = (await jwt_decode(Authorization.split(' ')[1])) as JwtPayload);
+    logger.info('getRoutesByUser user identifier included', { data: { username } });
+  }
   const Items = userRouteIndexItems.map((item) => {
-    return { ...item, gymName: gymLocationToNameMapping[item.gymLocation] };
+    let hasVoted = false;
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (qwoprocks) [inline lambda_backend/route_microservice/src/getRoutesByUser.ts:80]: Is this needed?

File: lambda_backend/route_microservice/src/getRoutesByUser.ts [hunk 2]
```diff
@@ -58,18 +62,37 @@ const getRoutesByUser: Handler = async (event: GetRoutesByUserEvent) => {
...
+    const { upvotes, ...returnItem } = item;
+    upvotes.forEach((name) => {
```
Comment (qwoprocks) [inline lambda_backend/route_microservice/src/getRoutesByUser.ts:82]: Would it be better to just use `Array.find()`?

File: lambda_backend/route_microservice/src/getRoutesByGym.ts [hunk 3]
```diff
@@ -27,17 +30,37 @@ const getRoutesByGym: Handler = async (event: GetRoutesByGymEvent) => {
...
+  let username = '';
+  if (Authorization) {
+    // Only utilizes token for username, no enforcing authorization as endpoint is GET & public
+    ({ username } = (await jwt_decode(Authorization.split(' ')[1])) as JwtPayload);
+    logger.info('getRoutesByGym user identifier included', { data: { username } });
+  }
+  const Items = gymLocationIndexItems.map((queryItem) => {
+    let hasVoted = false;
+    const { upvotes, ...returnItem } = queryItem;
+    upvotes.forEach((name) => {
```
Comment (qwoprocks) [inline lambda_backend/route_microservice/src/getRoutesByGym.ts:51]: `.find()`?

**Review by qwoprocks — APPROVED** [review]
LGTM

File: lambda_backend/route_microservice/src/getRoutesByUser.ts (hunk 2 above)
Comment (yarkhinephyo) [reply lambda_backend/route_microservice/src/getRoutesByUser.ts:82]: `code quality: A`

**Pull request merged**